src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/StateMachineFieldSymbol.cs

```diff
@@ -83,4 +85,34 @@ internal override bool IsCapturedFrame
             get { return _isThis; }
         }
     }
+
+    internal sealed class StateMachineFieldSymbolForRegularParameter : StateMachineFieldSymbol
+    {
+        private readonly ParameterSymbol _parameter;
+
+        public StateMachineFieldSymbolForRegularParameter(NamedTypeSymbol stateMachineType, TypeWithAnnotations type, string name, ParameterSymbol parameter, bool isPublic)
+            : base(stateMachineType, type, name, isPublic, isThis: false)
+        {
+            Debug.Assert(parameter is { IsThis: false });
+            _parameter = parameter;
+        }
+
+        internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<CSharpAttributeData> attributes)
+        {
+            if (_parameter.OriginalDefinition is SourceParameterSymbolBase definition &&
+                ContainingModule == definition.ContainingModule)
+            {
+                foreach (CSharpAttributeData attr in definition.GetAttributes())
+                {
+                    if (attr.AttributeClass is { HasCompilerLoweringPreserveAttribute: true } attributeType &&
+                        (attributeType.GetAttributeUsageInfo().ValidTargets & System.AttributeTargets.Field) != 0)
+                    {
+                        AddSynthesizedAttribute(ref attributes, attr);
+                    }
+                }
```

Comment:
This block to copy attributes is repeated in several locations. Would it make sense to extract a helper method?